### PR TITLE
Fix mice code and formatting in lecture 6

### PR DESCRIPTION
**Pull request by QuLogic**
Headings need a space after `#`. The broken mice code was a small typo: `ppm` -> `pmm`.

I don't think the extra backticks did any harm; this is just for consistency.

### CODE REVIEW

**Comment by QuLogic** [comment]
Ugh, stupid global namespace.

**Comment by lwjohnst86** [comment]
"stupid global namespace" I think it's less about namespace and more about improper package creation by authors of mice etc. This is an area I think Python is really good at... since you need to specify the package that the function is coming from (`npm.function_name`) so you would never encounter this problem... but R doesn't have this. Unfortunate :(

File: lec06-exploratory-data-analysis.Rmd [hunk 1]
```diff
@@ -210,73 +210,73 @@ ggplot(data = diamonds2, aes(x = x, y = y)) +
...
+meth <- init$method
+predM <- init$predictorMatrix
 ```
 
 To impute the missing values, MICE package uses an algorithm in a such a way that information from all the other variables in the dataset is used to predict and impute the missing values. Therefore, you may not want to use a certain variable as a predictor. For example, the ID variable does not have any predictive value, so we want to remove it. 
 
 The code below will remove a variable as a predictor in imputing missing values, but will not stop MICE from imputing values for that variable itself.
 ```{r}
-# predM[, c("record_id")]=0
+predM[, c("record_id")]=0
```
Comment (lwjohnst86) [inline lec06-exploratory-data-analysis.Rmd]: = to <-

File: lec06-exploratory-data-analysis.Rmd [hunk 2]
```diff
@@ -210,73 +210,73 @@ ggplot(data = diamonds2, aes(x = x, y = y)) +
...
 ```
 
 By default mice() includes all the variables in the data; it is suggested that we use as many variables as possible. Do not exclude a variable as predictor from the imputation model if that variable is going be included in your final analysis model.
 
 
 If you want to skip a variable from imputation itself use the code below. This variable will still be used for prediction. Let's exclude all the variables where we aren't missing data or where imputation isn't of interest:
 ```{r}
-# meth[c("year", "day", "month", "plot_id", "species_id", "plot_type", "genus", "species", "taxa")]=""
+meth[c("year", "day", "month", "plot_id", "species_id", "plot_type", "genus", "species", "taxa")]=""
```
Comment (lwjohnst86) [inline lec06-exploratory-data-analysis.Rmd]: <-

File: lec06-exploratory-data-analysis.Rmd [hunk 3]
```diff
@@ -210,73 +210,73 @@ ggplot(data = diamonds2, aes(x = x, y = y)) +
...
 ```{r}
-# init$meth
+init$meth
 ```
 
-The variables ``hindfoot_length`` and ``weight`` are already set to ppm, so we will leave it like that. However, if we wanted to directly specify the method, we could use the following code:
+The variables `hindfoot_length` and `weight` are already set to pmm, so we will leave it like that. However, if we wanted to directly specify the method, we could use the following code:
 ```{r}
-# meth[c("hindfoot_length")]="ppm"
+meth[c("hindfoot_length")]="pmm"
```
Comment (lwjohnst86) [inline lec06-exploratory-data-analysis.Rmd]: <-

File: lec06-exploratory-data-analysis.Rmd [hunk 4]
```diff
@@ -210,73 +210,73 @@ ggplot(data = diamonds2, aes(x = x, y = y)) +
...
 ```
 
 Now it is time to run the multiple (m=5; this is generally standard) imputation.
 ```{r}
-# set.seed(103)
-# imputed = mice(survey, method=meth, predictorMatrix=predM, m=5)
+set.seed(103)
+imputed = mice(survey, method=meth, predictorMatrix=predM, m=5)
```
Comment (lwjohnst86) [inline lec06-exploratory-data-analysis.Rmd]: <-

**Comment by QuLogic** [comment]
I mean the global namespace when rmarkdown generates a site, and keeps the namespace around for subsequent files.

File: lec06-exploratory-data-analysis.Rmd [hunk 5]
```diff
@@ -286,36 +286,40 @@ method – Refers to method used in imputation. We used predictive mean matching
...
+set.seed(10)
+iris2[sample(1:nrow(iris2), 20), "Sepal.Length"] <- NA
+iris2[sample(1:nrow(iris2), 20), "Petal.Length"] <- NA
 ```
 
-
+```{r, show=FALSE}
+unloadNamespace("mice")
+unloadNamespace("mitml")
+unloadNamespace("jomo")
```
Comment (lwjohnst86) [inline lec06-exploratory-data-analysis.Rmd:324]: just remove this line

**Review by lwjohnst86 — CHANGES_REQUESTED** [review]

File: lec06-exploratory-data-analysis.Rmd (hunk 5 above)
Comment (QuLogic) [reply lec06-exploratory-data-analysis.Rmd:324]: Can't; [breaks lec9](https://travis-ci.org/UofTCoders/rcourse/builds/433090310#L2870).

**Pull request merged**